### PR TITLE
enforce_home_region_follower_reads_enabled session var

### PR DESCRIPTION
**Pull request by taroface**
DOC-6784

Document the `enforce_home_region_follower_reads_enabled` session setting.

### CODE REVIEW

File: _includes/v23.1/misc/session-vars.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
...
+| <a name="enforce-home-region-follower-reads-enabled"></a> `enforce_home_region_follower_reads_enabled` | If `on`, allows the [`enforce_home_region`](cost-based-optimizer.html#control-whether-queries-are-limited-to-a-single-region) setting to perform `AS OF SYSTEM TIME` [follower reads](follower-reads.html) to determine a query's [home region](multiregion-overview.html#table-localities), if any. | `off` | Yes | Yes |
```
Comment (msirek) [inline _includes/v23.1/misc/session-vars.md:25]: Maybe change this to be more similar to https://github.com/cockroachdb/cockroach/blob/3042c4fee9457b4f05318fcfa9836433203430cb/pkg/sql/sessiondatapb/local_only_session_data.proto#L346-L348

Either use "determine and report" or "detect and report", or something similar in place of "determine", and also mention that `enforce_home_region` must also be set to `on` for this setting to have any effect.

**Review by msirek — CHANGES_REQUESTED** [review]

File: _includes/v23.1/misc/session-vars.md (hunk 1 above)
Comment (msirek) [reply _includes/v23.1/misc/session-vars.md:25]: Also, would this be the place to mention that `enforce_home_region_follower_reads_enabled` is a preview feature?

Comment (taroface) [reply _includes/v23.1/misc/session-vars.md]: Done and done - thanks for pointing out that this is in preview! PTAL.

**Review by msirek — APPROVED** [review]

**Review by kathancox — APPROVED** [review]
LGTM!

**Pull request merged**